src/compiler.ts

```diff
@@ -6343,16 +6343,18 @@ export class Compiler extends DiagnosticEmitter {
     expr: ExpressionRef,
     type: Type
   ): ExpressionRef {
+    let closureContextLocal = this.currentFlow.getTempLocal(Type.i32)
```

Comment:
Looks like the local doesn't become free'd again.

Reply:
Is freeing locals necessary for functionality? I was under the impression that it was just an optimization to use fewer locals- I figured for now I'd just omit them until I get this compiling/validating/optimized and return to it

Reply:
Not strictly for functionality, but other code does this so we don't end up with huge untouched output.

Reply:
Agreed that this is needed before merging- but for now I think I'll wait until I get get into a stable state (compiles, validates, optimizes successfully) before looking at this again unless you think that we need it sooner